modules/phase_field/src/utils/PolycrystalICTools.C

```diff
@@ -297,45 +300,78 @@ PolycrystalICTools::buildNodalGrainAdjacencyGraph(
       if (my_grain != their_grain)
         /**
          * We've found a grain neighbor interface. In order to assign order parameters though, we
-         * need to make
-         * sure that we build out a small buffer region to avoid literal "corner cases" where nodes
-         * on opposite
-         * corners of a QUAD end up with the same OP because those nodes are not nodal neighbors.
-         *
-         * To do that we'll build a halo region based on these interface nodes. For now, we need to
-         * record
-         * the nodes inside of the grain and those outside of the grain.
+         * need to make sure that we build out a small buffer region to avoid literal "corner cases"
+         * where nodes on opposite corners of a QUAD end up with the same OP because those nodes are
+         * not nodal neighbors. To do that we'll build a halo region based on these interface nodes.
+         * For now, we need to record the nodes inside of the grain and those outside of the grain.
          */
-        adjacency_matrix[my_grain][their_grain] = 1;
+        adjacency_matrix(my_grain, their_grain) = 1;
     }
   }
 
   return adjacency_matrix;
 }
 
 std::vector<unsigned int>
-PolycrystalICTools::assignOpsToGrains(const AdjacencyGraph & adjacency_matrix,
+PolycrystalICTools::assignOpsToGrains(AdjacencyMatrix<Real> & adjacency_matrix,
                                       unsigned int n_grains,
-                                      unsigned int n_ops)
+                                      unsigned int n_ops,
+                                      const MooseEnum & coloring_algorithm)
 {
   Moose::perf_log.push("assignOpsToGrains()", "PolycrystalICTools");
 
   std::vector<unsigned int> grain_to_op(n_grains, GraphColoring::INVALID_COLOR);
 
-  if (!colorGraph(adjacency_matrix, grain_to_op, n_grains, n_ops, 0))
-    mooseError(
-        "Unable to find a valid Grain to op configuration, do you have enough op variables?");
+  // Use a simple backtracking coloring algorithm
```

Comment:
For now at least. It does a better job of distributing the OPs among the variables for one thing. It works fine for just about any 2D problem. It also works without PETSc (which I know isn't very important). I'll also want to run comparisons between this algorithm and some of the more advanced ones.